src/main/java/seedu/address/model/person/Person.java

```diff
@@ -131,6 +131,16 @@ public boolean isSamePerson(Person otherPerson) {
                 && otherPerson.getName().equals(getName());
     }
 
+
+    /**
+     * Create a new person object to ensure immutability
+     * @return a copy of person
+     */
+    public Person copy() {
+        return new Person(this.name, this.phone, this.email, this.address, this.payment, this.participationList,
+                this.tags);
```

Comment:
Isn't participationList the same object?
The name, phone etc. are immutable so those are fine. But the participationList is a List that we can edit.

Reply:
Let me update this

Reply:
I created a shallow copied list using new ArrayList<>(this.participationList);